src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -176,6 +178,19 @@ struct patcher
     bd_data_ptr[2] = (bd_data_ptr[2] & 0xFFFF0000) | (base_address >> 32);            // NOLINT
   }
 
+  void patch_tansaction_ctrlpkt_48(uint32_t* bd_data_ptr, uint64_t patch)
+  {
+    uint64_t val = *reinterpret_cast<uint64_t*>(bd_data_ptr);
+    val &= 0x0000FFFFFFFFFFFF;                                                        // NOLINT
+    val += patch;
+    auto lower32BitVal = static_cast<uint32_t>(val);
+    auto higher16BitVal = static_cast<uint16_t>((val & 0x0000FFFF00000000) >> 32);    // NOLINT
+    *bd_data_ptr = lower32BitVal;
+
+    auto pHigher16Bit = reinterpret_cast<uint16_t*>(bd_data_ptr + 1);
+    *pHigher16Bit = higher16BitVal;
+  }
```

Comment:
I think the key here is by doing this patching, we are actually define the bd_data_ptr is pointing to a BD instead of pointing to the beginning of a control packet. My suggestion is the patch schema is based on the hardware architecture but not dpu sequence and transaction buffer. 

My thought on this is we should be able to reuse patch_ctrl48 and patch_shim48 for both DPU sequence and transaction buffer

Reply:
"Reuse patch_ctrl48 and patch_shim48 for both DPU sequence and transaction buffer" is an optimization.
In order to achieve that, we need to modify assembler.
Refer to this PR: https://gitenterprise.xilinx.com/XRT/aiebu/pull/16
It would be nice if "Reusing patch_ctrl48 and patch_shim48 for both DPU sequence and transaction buffer" was brought up there. And we do not need this new enum in this [line](https://gitenterprise.xilinx.com/XRT/aiebu/blob/ad0d57f9b6b66bdd83ed9be0835e2e2207ee7c70/src/python/aiebu/scripts/symbol.py#L23).

With that all being said, can we use what we have, optimize in another PR?

Reply:
I would not say it is an optimization because it is an design option. With this PR the enum is not clean. We should have 
dpu_patch_ctrl48
transaction_patch_ctrl48
if we want to two set of patching schema.

But if we add some comment explaining this is a temporary solution before we fix the assembler, and the code extra enum will be removed, I am fine to merge it for now.

Reply:
Comment added